pages/pil/training/routers/modules.js

```diff
@@ -31,6 +31,15 @@ module.exports = settings => {
       res.locals.static.schema = schema;
       res.locals.static.modulesThatRequireSpecies = modulesThatRequireSpecies;
       next();
+    },
+    process: (req, res, next) => {
+      req.form.values.modules.map(m => {
+        const specs = Object.values(pickBy(req.body, (value, key) => {
+          return startsWith(key, `module-${m}-species`);
+        })).filter(s => s !== '');
+        req.form.values[`module-${m}-species`] = specs;
+      });
+      next();
```

Comment:
could we extract this logic into a shared middleware?

Reply:
done